### ISSUE TITLE
GI: Cannot select directionality when overexpressed alleles are present

### ISSUE THREAD

**Issue by manulera** [pombase/canto#2615]
@kimrutherford @ValWood

The full list of constrains is in this issue: https://github.com/pombase/canto/issues/2593

Related to #2613 

## Affecting GIs as defined by BioGRID right now

Phenotypic suppression and enhancement should support genotypes in which either or both alleles are overexpressed / knockdown. Now the directionality is inferred the same way that for the GIs that start with `Dosage X`, assuming that the single allele phenotype is the one not overexpressed (or the first overexpressed if both are). Instead, in phenotypic enhancement / suppression, it should always be possible to select directionality.

## Affecting GIs outside of the current scope

If we interpret the definitions of BioGrid loosely (that's what they suggested in the email), then we could also have:

* A `Dosage X` (rescue, growth defect, lethality), in which the single allele phenotype is an overexpression. E.g overexpression of gene X causes lethality in a background where Y is overexpressed.
* A `Synthetic rescue`, in which the single allele phenotype is an overexpression. E.g. deletion of gene X causes letahlity in a background where Y is overexpressed.

**Comment by kimrutherford** [comment]
> Instead, in phenotypic enhancement / suppression, it should always be possible to select directionality.

Just to be clear, do we need to only change the case where both allele are overexpressed?  And keep the single overexpressed allele case the same?

**Comment by kimrutherford** [comment]
> And keep the single overexpressed allele case the same?

I mean, should we keep the functionality the same as it currently is for the single overexpressed allele case.

**Comment by kimrutherford** [comment]
I think I've fixed the code (and test Canto) so that if there are two overexpressed alleles, the user needs to select a direction for the interaction.  Please have a test when you get a chance and let me know if I got it right.  Thanks!

**Comment by manulera** [comment]
> I think I've fixed the code (and test Canto) so that if there are two overexpressed alleles

Hello Kim. I tested the case where both alleles are overexpressed, and it gives the expected behaviour. You can choose directionality and the suggested phenotypes are correct.

**Comment by manulera** [comment]
> > Instead, in phenotypic enhancement / suppression, it should always be possible to select directionality.
> 
> Just to be clear, do we need to only change the case where both allele are overexpressed? And keep the single overexpressed allele case the same?

Regarding this. In phenotypic suppression and phenotypic enhancement it should always be possible to select the directionality, regardless of either allele being overexpressed.

**Comment by kimrutherford** [comment]
> Regarding this. In phenotypic suppression and phenotypic enhancement it should always be possible to select the directionality, regardless of either allele being overexpressed.

Thanks Manu.  I think I've fixed that now.

**Comment by manulera** [comment]
Yes, looks good

**Comment by manulera** [comment]
It seems that I had not thought about this carefully enough.

> If we interpret the definitions of BioGrid loosely (that's what they suggested in the email) [...]
> A `Dosage X` (rescue, growth defect, lethality), in which the single allele phenotype is an overexpression. E.g overexpression of gene X causes lethality in a background where Y is overexpressed.

The relevant lines in the canto.yaml are these:

https://github.com/pombase/canto/blob/dc2468c17c8888f3a14c3162939648c1f5c04090/canto.yaml#L1710-L1735

`Dosage Rescue` works OK, because it does not have `interaction_does_not_need_phenotype: 1`, which forces you to pick directionality. Also, it makes sense because it is necessarily asymmetric: the single-locus strain has a phenotype that is rescued in the double mutant.

The problem is for `Dosage Growth defect` and `Dosage lethality`. Sticking strictly to BioGrid definitions, it should only be possible to use multi-locus genotypes in which one allele is overexpressed and the other isn't. Therefore, the double mutant phenotypes are synthetic (because they do not occur in the single locus phenotype), but the genetic interaction is directional (because only one is overexpressed). However, we relaxed that constraint and allowed genotypes where both alleles are overexpressed.

The canto config (rightly so) has `overexpression_implies_direction: 1` for these, but we are allowing them for strains in which two genes are overexpressed. Therefore, Canto picks one at random for the directionality. The thing is that there is no way to tell the directionality in this case, since both are overexpressed.

In summary: we should not allow `Dosage Growth defect` and `Dosage lethality` for genotypes that have two overexpression alleles.

**Comment by ValWood** [comment]
@kimrutherford Do we have any examples of  dosage growth  or dosage lethality where both are overexpression?

**Comment by kimrutherford** [comment]
> Do we have any examples of dosage growth or dosage lethality where both are overexpression?

I can't find any with a Chado query.

**Comment by ValWood** [comment]
I think that makes sense...you would not have done this in the old interface because you 
you would need to say
OEX a is rescued by OEX of b (which would be odd).

And here it would be strange too
<img width="882" alt="Screenshot 2023-09-03 at 07 54 05" src="https://github.com/pombase/canto/assets/7359272/3d728d5c-a3d8-4410-8bc0-6e223f62b5ed">


So we can block it as an option, but it is unlikely that users will try to do it so it probably isn't urgent
